### PR TITLE
Deprecate and replace withTimeoutMillis

### PR DESCRIPTION
**Pull request by spkrka**

### CODE REVIEW

File: folsom/src/main/java/com/spotify/folsom/MemcacheClientBuilder.java [hunk 1]
```diff
@@ -399,14 +399,27 @@ private void updateResolver() {
...
   public MemcacheClientBuilder<V> withRequestTimeoutMillis(final long timeoutMillis) {
-    this.timeoutMillis = timeoutMillis;
+    return withConnectionTimeoutMillis(timeoutMillis);
+  }
+
+  /**
+   * Set the maximum time to wait before considering the connection to be dead and should be
+   * closed and recreated.
+   *
+   * Do not use this to enforce request timeouts
```
Comment (protocol7) [inline folsom/src/main/java/com/spotify/folsom/MemcacheClientBuilder.java]: Should we give a clue on what to do instead?

**Review by rschildmeijer — APPROVED** [review]

**Review by c4pone — APPROVED** [review]

File: folsom/src/main/java/com/spotify/folsom/MemcacheClientBuilder.java (hunk 1 above)
Comment (spkrka) [reply folsom/src/main/java/com/spotify/folsom/MemcacheClientBuilder.java]: Added to readme - add something here too?

File: README.md [hunk 2]
```diff
@@ -139,6 +139,21 @@ If you are still on Java 7, you can depend on the older version:
...
 
+### Best practices
+
+Do not use `withConnectionTimeoutMillis()` or the deprecated `withRequestTimeoutMillis()` to set timeouts per request.
+This is intended to detect broken TCP connections to close it and recreate it. Once this happens, all open
+request will be completed with a failure and Folsom will try to recreate the connection. If this timeout is set too low,
+this will create connection flapping which will result in an increase of failed requests and/or increased
+request latencies.
+
+A better way of setting timeouts on individual requests (In Java 9+) is something like this:
```
Comment (protocol7) [inline README.md]:
```suggestion
A better way of setting timeouts on individual requests (in Java 9+) is something like this:
```

File: README.md [hunk 3]
```diff
@@ -139,6 +139,21 @@ If you are still on Java 7, you can depend on the older version:
   race condition bugs and deadlocks. We try to isolate that as much as possible to minimize the risk,
   and most of the code base doesn't have to care.
 
+### Best practices
+
+Do not use `withConnectionTimeoutMillis()` or the deprecated `withRequestTimeoutMillis()` to set timeouts per request.
+This is intended to detect broken TCP connections to close it and recreate it. Once this happens, all open
+request will be completed with a failure and Folsom will try to recreate the connection. If this timeout is set too low,
```
Comment (protocol7) [inline README.md]:
```suggestion
requests will be completed with a failure and Folsom will try to recreate the connection. If this timeout is set too low,
```

File: README.md [hunk 4]
```diff
@@ -139,6 +139,21 @@ If you are still on Java 7, you can depend on the older version:
...
+this will create connection flapping which will result in an increase of failed requests and/or increased
+request latencies.
+
+A better way of setting timeouts on individual requests (In Java 9+) is something like this:
+
+```java
+CompletableFuture<T> future = client.get(...).toCompletableFuture().orTimeout(...);
+
```
Comment (protocol7) [inline README.md]:
```suggestion
```

**Review by protocol7 — APPROVED** [review]

**Pull request merged**